### PR TITLE
Add lldb flag to executables

### PR DESCRIPTION
**Pull request by OmarElawady**
Add `--lldb` flag to support debugging for executable files in lldb similar to gdb.
An entry is added to the help message for the new flag `--lldb`.
The code to run the file in the debugger is the same as for gdb with slight modifications to work with lldb.

### CODE REVIEW

**Comment by OmarElawady** [comment]
Added a test for LLDB launch.

For some reason LLDB doesn't commands to be piped into the debugger. When running something like `echo run | lldb ./someapp` or lldb ./someapp < commands.txt, it launches the processes and halts immediately without executing it. That's why I wasn't able to add tests to check the whether the executable is launched properly with the correct arguments (although I tested it manually and it worked as expected).

File: runtime/include/chpl-comm.h [hunk 1]
```diff
@@ -190,6 +190,15 @@ void chpl_comm_post_mem_init(void);
 //
 int chpl_comm_run_in_gdb(int argc, char* argv[], int gdbArgnum, int* status);
 
+//
+// if possible, run in lldb (because the user threw the --lldb flag)
+// using argc and argv.  gdbArgnum gives the index of the argv[]
```
Comment (bradcray) [inline runtime/include/chpl-comm.h]: `gdbArgNum` needs to be updated here.

**Comment by bradcray** [comment]
It looks to me as though the PRECOMP file is not needed?

Other than those two minor comments, this looks good to me, and works as I'd hoped.  Thanks!

**Comment by OmarElawady** [comment]
@bradcray Thanks for reviewing. I removed PRECOMP and fixed `gdbArgnum`.

**Comment by bradcray** [comment]
Looks great, thanks @OmarElawady!!

**Comment by bradcray** [comment]
Thanks again for taking this on, @OmarElawady.  I'm already benefitting from it and grateful that it's there!

**Comment by OmarElawady** [comment]
@bradcray Thank you. I'm very glad you found it useful.

**Pull request merged**